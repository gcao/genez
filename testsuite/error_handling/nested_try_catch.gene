# Test nested try/catch blocks
(try
  (do
    (println "Outer try")
    (try
      (do
        (println "Inner try")
        (throw "Inner error"))
      (catch e
        (do
          (println "Inner caught:" e)
          (throw "Rethrow")))))
  (catch e
    (println "Outer caught:" e)))