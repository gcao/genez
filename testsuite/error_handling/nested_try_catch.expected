Outer try
Inner try
Inner caught: Inner error
Outer caught: Rethrow